### PR TITLE
Minor code refactor aimed at optimization code

### PR DESCRIPTION
**Pull request by GermanAizek**
@duncanspumpkin, @Gymnasiast review code and feedback please.

### CODE REVIEW

File: src/openrct2/core/FileWatcher.cpp [hunk 1]
```diff
@@ -82,9 +82,11 @@ FileWatcher::WatchDescriptor::~WatchDescriptor()
 #endif
 
 FileWatcher::FileWatcher(const std::string& directoryPath)
+#ifdef _WIN32
+    : _path(directoryPath)
+#endif
 {
 #ifdef _WIN32
-    _path = directoryPath;
```
[the remark this answers is not in the document]
Comment (GermanAizek) [reply src/openrct2/core/FileWatcher.cpp]: @Broxzier fix it.
https://github.com/OpenRCT2/OpenRCT2/pull/17055/commits/77135bf28faaab3573ba4aceb32fc3152cca5400

File: src/openrct2/network/Socket.cpp [hunk 2]
```diff
@@ -388,11 +388,11 @@ class TcpSocket final : public ITcpSocket, protected Socket
 
                 if (rc == 0)
                 {
-                    tcpSocket = std::unique_ptr<ITcpSocket>(new TcpSocket(socket, hostName, ipAddress));
+                    tcpSocket = std::make_unique<ITcpSocket>(TcpSocket(socket, hostName, ipAddress));
```
Comment (Broxzier) [inline src/openrct2/network/Socket.cpp:391]:
```suggestion
                    tcpSocket = std::make_unique<TcpSocket>(socket, hostName, ipAddress);
```

**Comment by GermanAizek** [comment]
@Broxzier, oh thx for commits.
Yes, accidentally wrote the wrong expression during code refactor.

File: src/openrct2/core/FileIndex.hpp [hunk 3]
```diff
@@ -204,7 +204,7 @@ template<typename TItem> class FileIndex
             JobPool jobPool;
             std::mutex printLock; // For verbose prints.
 
-            std::list<std::vector<TItem>> containers;
+            std::vector<std::vector<TItem>> containers;
```
Comment (Gymnasiast) [inline src/openrct2/core/FileIndex.hpp:207]: What is the reason behind this change?

**Review by Gymnasiast — COMMENTED** [review]
Code looks good, although you’ll need to fix compilation. (Don’t worry too much about the commit message, we can fix that during squash merge this time.)

I have one question, which I left as a line note.

File: src/openrct2/core/FileIndex.hpp (hunk 3 above)
Comment (Broxzier) [reply src/openrct2/core/FileIndex.hpp:207]: Internally, a vector is a pointer to a buffer. Having these pointers close to each other is a micro-optimization for when iterating over them.

**Comment by Broxzier** [comment]
@GermanAizek You'll either need to reintroduce the `new` as argument to the `std::unique_ptr`'s constructor insteaf of using `std::make_unqiue`, or make `TcpSocket`'s constructor that is used public. It was likely made private on purpose, though I don't really see the reason to.

**Comment by Broxzier** [comment]
Reverted the second commit, and now CI passes again. Something in that commit must've broken something. I don't spot anything obvious, so I'm going to reintroduce and push each change separately.
